examples/src/test/scala/docs/kafka/KafkaDocExample.scala

```diff
@@ -58,6 +67,55 @@ object KafkaDocExample {
   }
   //#handler
 
+  //#wordSource
+  final case class WordEnvelope(offset: Long, word: Word)
+
+  class WordSource(implicit ec: ExecutionContext) extends SourceProvider[Long, WordEnvelope] {
+
+    private val words = Vector("aaaa", "bbbb", "cccc", "dddd", "eeee", "ffff", "gggg")
+    private val rnd = new Random(17) // same sequence each time
+    private val src = Source
+      .fromIterator(() =>
+        new Iterator[WordEnvelope] {
+          private var offset = 0L
+          override def hasNext: Boolean = true
+
+          override def next(): WordEnvelope = {
+            val i = rnd.nextInt(words.size)
+            offset += 1
+            WordEnvelope(offset, words(i))
+          }
+        })
+      .throttle(1, 1.second)
+
+    override def source(offset: () => Future[Option[Long]]): Future[Source[WordEnvelope, _]] = {
+      offset().map {
+        case Some(o) => src.dropWhile(_.offset <= o)
+        case _       => src
+      }
+    }
+
+    override def extractOffset(env: WordEnvelope): Long = env.offset
+  }
+  //#wordSource
+
+  //#wordPublisher
+  class WordPublisher(topic: String, sendProducer: SendProducer[String, String])(implicit ec: ExecutionContext)
+      extends SlickHandler[WordEnvelope] {
+    private val logger = LoggerFactory.getLogger(getClass)
+
+    override def process(envelope: WordEnvelope): DBIO[Done] = {
+      val word = envelope.word
+      val partition = math.abs(word.hashCode % 3)
+      val key = if (word.isEmpty) "" else word.take(1)
+      val producerRecord = new ProducerRecord(topic, partition, key, word)
```

Comment:
I think it would be better to omit the partition selection and just rely on the `DefaultPartitioner`.  

```
val producerRecord = new ProducerRecord(topic, key, word)
```

This will be particularly important if we expand this example, or add new examples that use the Alpakka Kafka external sharding support where the default implementation relies on the `DefaultPartitioner`.

If you want to include the partition in the log message below you could map the future returned by `SendProducer.send` and retrieve it from the `RecordMetadata`.

Reply:
Sounds good, I'll change. Is that based on the `key`? Do you think I should include the full word as the key in this example?

Reply:
Yes, the `DefaultPartitioner` will use the `key`.

> Do you think I should include the full word as the key in this example?

Since there's no expectation of ordering because there's no consumer example I guess it doesn't matter.  I think it would make sense to use the whole word as the key if you make up a rationale for that on what could be consuming it downstream.

Reply:
The example is about word counting, so nice if a word always ends up in same partition and same consuming projection instance. Changed in 64f7402